v2/api/apimanagement/v1api20220801/structure.txt

```diff
@@ -206,6 +206,87 @@ github.com/Azure/azure-service-operator/v2/api/apimanagement/v1api20220801
 │           ├── "Header"
 │           ├── "Query"
 │           └── "Segment"
+├── AuthorizationProvider: Resource
+│   ├── Owner: Service
+│   ├── Spec: Object (5 properties)
+│   │   ├── AzureName: Validated<string> (3 rules)
+│   │   │   ├── Rule 0: MaxLength: 256
+│   │   │   ├── Rule 1: MinLength: 1
+│   │   │   └── Rule 2: Pattern: "^[^*#&+:<>?]+$"
+│   │   ├── DisplayName: Validated<*string> (2 rules)
+│   │   │   ├── Rule 0: MaxLength: 300
+│   │   │   └── Rule 1: MinLength: 1
+│   │   ├── IdentityProvider: *string
+│   │   ├── Oauth2: *Object (2 properties)
+│   │   │   ├── GrantTypes: *Object (2 properties)
+│   │   │   │   ├── AuthorizationCode: *genruntime.SecretMapReference
+│   │   │   │   └── ClientCredentials: map[string]string
+│   │   │   └── RedirectUrl: *string
+│   │   └── Owner: *genruntime.KnownResourceReference
+│   └── Status: Object (7 properties)
+│       ├── Conditions: conditions.Condition[]
+│       ├── DisplayName: *string
+│       ├── Id: *string
+│       ├── IdentityProvider: *string
+│       ├── Name: *string
+│       ├── Oauth2: *Object (2 properties)
+│       │   ├── GrantTypes: *Object (2 properties)
+│       │   │   ├── AuthorizationCode: map[string]string
+│       │   │   └── ClientCredentials: map[string]string
+│       │   └── RedirectUrl: *string
+│       └── Type: *string
+├── AuthorizationProvidersAuthorization: Resource
+│   ├── Owner: AuthorizationProvider
+│   ├── Spec: Object (7 properties)
+│   │   ├── AuthorizationType: *Enum (1 value)
+│   │   │   └── "OAuth2"
+│   │   ├── AzureName: Validated<string> (3 rules)
+│   │   │   ├── Rule 0: MaxLength: 256
+│   │   │   ├── Rule 1: MinLength: 1
+│   │   │   └── Rule 2: Pattern: "^[^*#&+:<>?]+$"
+│   │   ├── Error: *Object (2 properties)
+│   │   │   ├── Code: *string
+│   │   │   └── Message: *string
+│   │   ├── Oauth2GrantType: *Enum (2 values)
+│   │   │   ├── "AuthorizationCode"
+│   │   │   └── "ClientCredentials"
+│   │   ├── Owner: *genruntime.KnownResourceReference
+│   │   ├── Parameters: map[string]string
+│   │   └── Status: *string
```

Comment:
Same question as error - is this field actually user-settable? If not, mark it readonly so it doesn't clutter the spec?